### PR TITLE
GRA-1205: internet gateways

### PR DESCRIPTION
**Pull request by 0xdcarns**
## Describe your changes
Internet gateways on **linux** only work with this PR
Will merge after Mac and Windows complete initial PRs

## Provide Issue ticket number if applicable/not in title
title.
## Provide link to Netmaker PR if required
- https://github.com/gravitl/netmaker/pull/2182
## Provide testing steps
- [x] Ensure that an egress gateway with 0.0.0.0/0 functions as an internet gateway for a connected linux peer

## Checklist before requesting a review
- [ ] My changes affect only 10 files or less.
- [x] I have performed a self-review of my code and tested it.
- [x] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [ ] If it is a bugfix, my code is <= 200 lines.
- [ ] My functions are <= 80 lines.
- [x] I have had my code reviewed by a peer.
- [x] My unit tests pass locally.
- [x] Netclient & Netmaker are awesome.

### CODE REVIEW

**Review by afeiszli — CHANGES_REQUESTED** [review]
In order to compile on Windows, should have unimplemented functions in routes_windows (see workflow failures)

**Comment by 0xdcarns** [comment]
> In order to compile on Windows, should have unimplemented functions in routes_windows (see workflow failures)

I state in PR won't be merged till other ones are in...

**Comment by theguy951357** [comment]
I still have nodes freezing up when I set an egress gateway to 0.0.0.0/0

**Review by theguy951357 — APPROVED** [review]
pr passes
critical list passes

File: config/config.go [hunk 1]
```diff
@@ -607,3 +615,92 @@ func Convert(h *Config, n *Node) (models.Host, models.Node) {
...
+
+	return isInetGW
+}
+
+func peerHasIp(ip *net.IPNet, allowedIPs []net.IPNet) bool {
+	if ip == nil {
+		return false
+	}
+	for i := range allowedIPs {
+		aIP := allowedIPs[i]
```
Comment (mattkasun) [inline config/config.go:679]: use 
```
func peerHasIp(ip *net.IPNet, allowedIPs []net.IPNet) bool {
    if ip.Contains(allowedIPs[i].IP) {
        return true
    }
    return false
}
```

**Review by mattkasun — CHANGES_REQUESTED** [review]

**Review by afeiszli — APPROVED** [review]

**Pull request merged**